CommandLine/emake/EnigmaCallbacks.cpp

```diff
@@ -3,33 +3,33 @@
 #include <iostream>
 #include <pthread.h>
 #include <unistd.h>
+#include <vector>
 
 std::ifstream CallBack::_outFile;
 
 void* CallBack::OutputThread(void*)
 {
-  if (_outFile)
-  {
+  if (_outFile.is_open()) {
     int pos = 0;
-    while (!_outFile.eof()) 
+    std::vector<char> buffer(size_t(4096));
+    while (_outFile.good() && !_outFile.eof()) 
     {
       _outFile.seekg(0, _outFile.end);
       int length = static_cast<int>(_outFile.tellg()) - pos;
       _outFile.seekg(pos, _outFile.beg);
  
       if (length > 0) 
       {
-        char* buffer = new char[length + 1];
-        _outFile.read(buffer, length);
+        _outFile.read(buffer.data(), length);
         buffer[length] = '\0';
         pos += length;
-        std::cout << buffer;
-        delete[] buffer;
+        std::cout << buffer.data();
       }
+      _outFile.clear();
     }
   }
-  
-  pthread_exit(nullptr);
+
+  return nullptr;
 }
```

Comment:
@JoshDreamland I originally told him to add ``pthread_exit(nullptr)`` but I had to do this to silence a ``no return statement`` warning. Is this correct or should you call both or what?